content/asm_6.md

```diff
@@ -1,181 +1,535 @@
-	.cfi_offset 6, -16
-	movq	%rsp, %rbp
-	.cfi_def_cfa_register 6
-	movl	$15, %edx
-	movl	$.LC0, %esi
-	movl	$1, %edi
-	call	write
-	movl	$0, %eax
-	popq	%rbp
-	.cfi_def_cfa 7, 8
-	ret
-	.cfi_endproc
-.LFE0:
-	.size	main, .-main
-	.ident	"GCC: (Ubuntu 4.9.1-16ubuntu6) 4.9.1"
-	.section	.note.GNU-stack,"",@progbits
+>>> 0.2 + 0.4
+0.6000000000000001
 ```
 
-Looks different then nasm Hello world, let's look on some differences.
+After we converted both integral and fractional parts of our number to binary representation, we can go to the next step. We need to convert to [scientific notation](https://en.wikipedia.org/wiki/Scientific_notation). The first part we need to [shift right](https://en.wikipedia.org/wiki/Logical_shift) the integral part of our number so that only `1` will remain. In a case of our number `101` - we need to shift right the number 2 times. As the result our initial number `101.101` started to be represented as `1.01101 * 2^2`. After this we need to add the number of shifts we done to the special number called - `bias`. For single-precision floating-point numbers it is equal to `127`. So we will get - `2 + 127 = 129` or `0b10000001`. This will be our `exponent`. The `mantissa` is just the fractional part of the number that we got after shifting it. We just put all the number of the fractional parts to the 23 `mantissa` bits.
 
-## AT&T syntax
+If we will combine all together, we may see how our number `5.625` will be represented in a computer memory:
 
-### Sections
+| Sign | Exponent | Mantissa                |
+|------|----------|-------------------------|
+|     0| 10000001 | 01101000000000000000000 |
 
-I don't know how about you, but when I start to write assembler program, usually I'm starting from sections definition. Let's look on simple example:
+If the fractional part of the number is periodic as we have seen with the example of the number `5.575` - we will fill the `mantissa` bits while it is possible. So the number `5.575` will be represented in a computer memory as:
 
-```assembly
-.data
-    //
-    // initialized data definition
-    //
-.text
-    .global _start
+| Sign | Exponent | Mantissa                |
+|------|----------|-------------------------|
+|     0| 10000001 | 01100100110011001100110 |
 
-_start:
-    //
-    // main routine
-    //
-```
+Now we know how computer represents a floating point number with a single-precision 🎉 
+
+### Double and double-extended precision formats
+
+As you may guess, the representation of the floating-point numbers using the double-precision and double-extended precision formats is similar to the single-precision format and the difference is only accuracy.
+
+The double-precision format is `64-bits` of memory with the following structure:
+
+- sign - 1 bit
+- exponent - 11 bit
+- mantissa - 52 bit
+
+The extended-double precision format is `80-bits` of memory with the following structure:
+
+- sign - 1 bit
+- exponent - 15 bit
+- mantissa - 112 bit
```

Comment:
```suggestion
- Sign - 1 bit
- Exponent - 15 bit
- Mantissa - 112 bit
```

Reply:
Are these values correct? Shouldn't they sum up to give 80? (that was the case for double-precision format).